templates/problem/voting-controls.html

```diff
@@ -0,0 +1,210 @@
+<style>
+    .vote-form {
+        background-color: rgb(255,255,255);
+        padding: 40px;
+        border-radius: 25px;
+        justify-content: center;
+        align-items: center;
+        margin: auto;
+        text-align: center;
+    }
+
+    .vote-form-info {
+        font-size: 24px;
+        font-weight: bold;
+        padding:10px
+    }
+
+    .vote-form-value {
+        font-size: 24px;
+    }
+
+    .vote-form-text {
+        font-size: 18px;
+    }
+
+    .voting-form-error {
+        font-size: 18px;
+        color: red;
+    }
+
+    .vote-button {
+        float:right;
+    }
+
+    .form-button {
+        cursor: pointer;
+    }
+</style>
+
+{% if not in_contest %}
+    <hr style="padding-top: 0.1em">
+{% endif %}
+
+{% include 'lightbox.html' %}
+
+{% if can_vote %}
+    <div class="vote-form" id="id_vote_form_box">
+        <div id="id_has_voted_prompt">
+            <span class="vote-form-info">{{ _('Current Vote:') }}</span>
+            <span id="id_current_vote_value" class="vote-form-value"></span>
+            <br/>
+            <br/>
+            <span class="vote-form-text">{{  _('Change your vote below!') }}</span>
+            <br/>
+            <br/>
+        </div>
+        <div id="id_has_not_voted_prompt">
+            <span class="vote-form-text">{{ _("Cast your vote on this problem's weightage below!") }}</span>
+            <br/>
+            <br/>
+        </div>
+        <form id="id_vote_form">
+            {% csrf_token %}
+            <span class="vote-form-info">{{ _('Points') }}</span>
+            <input id="id_points" class="vote-form-value" type="number" step="any" name="points">
+            <script>
+                $('#id_points').keydown(e => {
+                    const invalidChars = ['e','+','-','.'];
```

Comment:
The number field automatically black lists most characters. This narrows that to only characters that would fit an integer.

Reply:
If you stick to doing this, use the [pattern](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern) attribute.

Reply:
Instead of this, you could set the [minimum and maximum for the number field.](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#specifying_minimum_and_maximum_values)

Reply:
Not really, that still lets you use every character in the `invalidChars` array. (`1e1`, `1.0`, `+1`, `-0`)

Reply:
Just let Django handle it server-side.

Reply:
it will display the error message of `this field is required` if I enter what @fluix-dev said into the field. Is that the intended message we want to say to the user?